routing.go

```diff
@@ -143,33 +143,80 @@ func (dht *IpfsDHT) GetValue(ctx context.Context, key string) ([]byte, error) {
 }
 
 func (dht *IpfsDHT) GetValues(ctx context.Context, key string, nvals int) ([]routing.RecvdVal, error) {
+	var resChan chan *routing.RecvdVal = make(chan *routing.RecvdVal, 0)
+	var errChan chan error = make(chan error, 0)
+	go dht.getValuesAsyncRoutine(ctx, key, nvals, resChan, errChan)
+
 	var vals []routing.RecvdVal
+loop:
+	for {
+		select {
+		case err := <-errChan:
+			if err == nil {
+				break loop
+			}
+			return nil, err
+		case res := <-resChan:
+			if res == nil {
+				break loop
+			}
+			vals = append(vals, *res)
+		}
+	}
+
+	return vals, nil
+}
+
+func (dht *IpfsDHT) GetValuesAsync(ctx context.Context, key string, nvals int) <-chan *routing.RecvdVal {
```

Comment:
You added the error field into the RecvdVal in the routing interface PR. Should we make use of it?